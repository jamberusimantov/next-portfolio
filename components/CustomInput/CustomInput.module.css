.input {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    margin: 0 auto;
    padding: 0 1rem;
    position: relative;
}

.input input {
    line-height: 2.5rem;
    font-size: large;
    width: 100%;
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
}

.input label {
    position: relative;
    top: 17px;
    right: 3px;
    left: 3px;
    font-size: small;
    color: Highlight;
}

.input span {
    position: relative;
    right: 0;
    left: 0;
    font-size: small;
    color: red;
}

@media(min-width:768px) {
    .input {
        width: 80%;
    }
}