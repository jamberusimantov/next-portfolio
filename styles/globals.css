html,
body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Oxygen, Ubuntu, Cantarell, Fira Sans, Droid Sans, Helvetica Neue, sans-serif;
}

a {
    color: inherit;
    text-decoration: none;
}

* {
    box-sizing: border-box;
}

.hamburger {
    width: 40px;
    height: 40px;
    border: 0 solid transparent;
    border-radius: 0;
    cursor: pointer;
    display: inline-block;
    padding: 5px;
    background-color: transparent;
}

.inner {
    margin: -1px 0 0;
    position: relative;
    width: 100%
}

.hamburger .bar {
    background-color: var(--white);
    border-radius: 0;
    content: "";
    display: block;
    height: 2px;
    position: absolute;
    width: 100%
}

.hamburger .bar:first-child {
    top: -10px;
}

.hamburger .bar:nth-child(3) {
    top: 10px;
}

.hamburger.active .bar:first-child {
    transform: rotate(45deg) translate3d(7px, 7px, 0);
    transition-delay: .1s
}

.hamburger.active .bar:nth-child(2) {
    transform: scaleX(0);
    transition-delay: .05s
}

.hamburger.active .bar:nth-child(3) {
    transform: rotate(-45deg) translate3d(7px, -7px, 0);
    transition-delay: .1s
}

.page {
    position: relative;
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: large;
}

.pageContainer {
    position: relative;
    min-height: calc(100vh - 4.5rem);
}

.response {
    height: 25px;
    width: 25px;
    padding: 0 2rem;
    background-repeat: no-repeat;
    background-position: center;
    background-size: contain;
}

.success {
    background-image: url('/assets/svg/success.svg');
}

.failure {
    background-image: url('/assets/svg/failure.svg');
}

.submit {
    background-image: url('/assets/svg/submit.svg');
}

.magnifier {
    background-image: url('/assets/svg/magnifier.svg');
}

.resetBtn,
.submitBtn {
    width: 150px;
}

.linkItem {
    display: inline-flex;
    flex-direction: row;
    align-items: center;
    color: var(--blue);
    margin: 0;
}

.white {
    color: var(--white) !important;
}

.linkItem:hover {
    cursor: pointer;
    text-decoration: underline;
}

.decoration {
    padding: 0 .2rem;
}

@media(min-width:768px) {
    .pageContainer {
        min-height: calc(100vh - 7rem);
    }
}