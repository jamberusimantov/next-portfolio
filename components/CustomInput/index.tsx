import React, { forwardRef } from "react";
import type { ChangeEventHandler, FocusEventHandler, CSSProperties } from "react";
import styles from './CustomInput.module.css'


interface iCustomInputProps {
    title: string;
    value?: string;
    description?: string;
    hasError?: boolean;
    required?: boolean;
    handleBlur?: FocusEventHandler<HTMLInputElement>;
    handleChange?: ChangeEventHandler<HTMLInputElement>;
    type?: keyof { "text": "", "email": "", "tel": "" };
    placeholder?: string;
    style?: CSSProperties;
}

const CustomInput = forwardRef<HTMLInputElement, iCustomInputProps>((props, ref) => {
    const { title, required, handleBlur, type = "text", placeholder, style, value, handleChange, description, hasError } = props;
    return (
        <div className={styles.input}>
            <label htmlFor={title} children={title} />
            <input
                ref={ref}
                title={title}
                name={title}
                value={value}
                onChange={handleChange}
                onBlur={handleBlur}
                type={type}
                required={required}
                placeholder={placeholder}
                style={style}
            />
            {hasError && required && <span children={" * " + (description ?? "")} />}
        </div>
    );
});

export default CustomInput;