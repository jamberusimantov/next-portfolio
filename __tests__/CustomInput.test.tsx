import React from "react";
import { describe, expect, test } from '@jest/globals';
import { create } from 'react-test-renderer';
import CustomInput from '../components/CustomInput';


describe('CustomInput', () => {
    test('renders correctly', () => {
        let tree = create(<CustomInput title="Language" />).toJSON();
        if (!tree) return;
        tree = tree instanceof Array ? tree[0] : tree;
        expect(tree).toMatchSnapshot();
    })
})