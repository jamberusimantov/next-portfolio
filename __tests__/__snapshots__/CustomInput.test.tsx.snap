// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomInput renders correctly 1`] = `
<div
  className="input"
>
  <label
    htmlFor="Language"
  >
    Language
  </label>
  <input
    name="Language"
    title="Language"
    type="text"
  />
</div>
`;
